core/scripts/generate.cjs

```diff
@@ -47,15 +47,23 @@ const generate = async () => {
   await generateSchema({
     input: getEndpoint(),
     headers: { Authorization: `Bearer ${getToken()}` },
-    output: join(__dirname, '../schema.graphql'),
+    output: join(__dirname, '../bigcommerce.graphql'),
     tsconfig: undefined,
   });
 
   await generateOutput({
     disablePreprocessing: false,
-    output: join(__dirname, '../graphql-env.d.ts'),
+    output: undefined,
     tsconfig: undefined,
   });
 };
 
-generate();
+(async () => {
+  try {
+    await generate();
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+    process.exit(1);
+  }
+})();
```

Comment:
Can you explain why this is needed?

Reply:
TL;DR:
1. The errors thrown in `getStoreHash` and `getToken` are synchronous
2. If these functions were called synchronously and an error was thrown, the script would immediately terminate (unless they were caught)
3. However, these functions are called within an asynchronous context (`generate`)
4. Since we are no longer synchronous, the `throw` statements emit an `unhandledRejection` event instead of immediately terminating the program

You can try this by reverting the code so that it looks like:
```ts
// ...

generate();
```

Then, remove a required environment variable from `.env.local` and try `pnpm generate` again. You'll notice the process exits with a `0` exit code, but no Schema or introspection files were generated. 

If you then add:

```ts
// ...

generate();

process.on('unhandledRejection', (error) => {
  console.error(error);
  process.exit(1);
});
```

And run `pnpm generate` again, you'll see the error logged to the console.

Reply:
Update: Refactored this to simply wrap the body of `generate` in a `try/catch`, lends to better readability compared to the IIFE